src/nix/key.rs

```diff
@@ -1,12 +1,18 @@
-use std::path::PathBuf;
+use std::{
+    io::{self, Cursor},
+    path::PathBuf,
+};
 
 use regex::Regex;
-use serde::{Serialize, Deserialize};
+use serde::{Deserialize, Serialize};
+use tokio::{fs::File, io::AsyncRead};
 use validator::{Validate, ValidationError};
 
 #[derive(Debug, Clone, Validate, Serialize, Deserialize)]
 pub struct Key {
-    pub(crate) text: String,
+    pub(crate) text: Option<String>,
```

Comment:
Could be a proper enum like:
```
Text(String),
KeyFile(String),
```
 but we would have to do more magic with serde

Reply:
Yeah, this is definitely nicer and will make it easier to implement #3 in a cleaner way.